proto/message_contents/proof.proto

```diff
@@ -0,0 +1,32 @@
+// Signature is a generic structure for public key signatures.
+syntax = "proto3";
+
+package xmtp.message_contents;
+
+import "message_contents/signature.proto";
+
+option go_package = "github.com/xmtp/proto/v3/go/message_contents";
+option java_package = "org.xmtp.proto.message.contents";
+
+// Association Proof proves that identity key material is associated with
+// a specific individually.
+message AssociationProof {
+    // Legacy: "XMTP : Create Identity ..." Signature Payload
+    message LegacyCreateIdentityProof {
+        // Text is the identitySignaturePayload e.g. Create Identity ...
+        bytes text = 1;
+        // Signature is the same as SignedPublicKey.Signature historically
+        Signature signature = 2;
```

Comment:
Could this be future proofed by adding the signature standard used in this case EIP191 at this level? OR do you think this is best left to a different workpackage?

My thoughts being, `Signature` object only captures how the signature is serialized not how it was generated. By adding the signing standard  EIP191, EIP1271, etc.

CAIP-74 seems like prior art here which the use of the SignatureMetaData. https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-74.md

Seems like there are 3 separate entities to consider.
- SignatureFormat - ECDSACompact
- SignatureGeneration - EIP191
- SignatureValidation - SIWE / XMTPCreate

Happy to discard if this complicates matters further.

Reply:
Sounds good, I can add these three fields, each of them is an enum right?

Reply:
SignatureFormat Is currently handled by the Signature type so that could be reused. 

The others would be enums.